stdlib/public/core/Flatten.swift.gyb

```diff
@@ -113,7 +113,7 @@ extension LazySequenceProtocol
 }
 
 %{
-def collectionForTraversal(traversal):
+def baseCollectionForTraversal(traversal):
```

Comment:
I actually prefer the old name.  Even in this file, we use this function not only for constraints on the base collection.

And this function is duplicated in multiple files anyway.  We would like to share it somehow one day, which would be easier if it had the same name everywhere.